contributing/COMMITS.md

```diff
@@ -0,0 +1,242 @@
+squash! [API] Add `/stores` endpoint             # Endpoint/Handlers/Domain implementation
+squash! [API] Add `/stores` endpoint             # Repository implementation
+[CLI] Add `store` noun and `add`/`delete` verbs  # Test + Implementation
+```
+
+In the above example, the larger chunk of work implementing an API has
+been split into several commits for TDD and readability. The commit body
+would describe the specifics of what has been done in that commit. These
+three commits would be squashed before merge, after review. The title of
+the first commit should reflect the final merged commit title.
+
+The CLI changes have been made as a single commit containing tests and
+implementation due to the simpler nature of that work. This commit would
+be left as a separate commit after merge, as adding the API, then
+binding it to the CLI are two logically separate things.
+
+The benefit here is that if the CLI commit needed to be rolled back due
+to a bug, it would then allow the API to remain functional (assuming
+there was no issue there).
+
+It's also worth noting that the contributor may have started with a
+completely different set of working commits before the Pull Request was
+made. The commits shown below were easily rearranged into the preferred
+sequence and granularity for review using an interactive git rebase.
+This allowed the 'end-to-end first' development approach to be used,
+without compromising the functionality of the release branch post merge.
+
+```
+[CLI] Add e2e test for store verbs
+[CLI] Add store verb commands
+[CLI] Add stores verb app service entry points
+[API] Add `/stores` to swagger spec
+[API] Add `/stores` e2e tests
+[API] Add `/stores` endpoint and handlers
+[API] Add `/stores` domain logic
+[API] Add `/stores` in mem repo
+[API] Add `/stores` sqlite repo
+[CLI] Connect store app service to API
+```
+
+> NOTE: If you do choose to use this approach, it is important to ensure
+> the full test suite passes at each final merge commit (post-squash)
+> after re-ordering.
+
+#### Making titles easier to read during review
+
+It's ok within a Pull Request to use other titles for squash commits to
+make keeping track of development easier. As long as it is still clear
+to the reviewer which commits will be squashed, and which are intended
+to be merged as-is. Use the `squash` prefix without the `!` to make it
+clear these need to be manually managed.
+
+The following example shows a situation where due to the number of
+changes required, the `--autosquash` format was avoided to make it
+easier to understand:
+
+```
+[API] Add `/replica` endpoint to swagger spec
+squash [API] Add tests for `/replica` endpoint
+squash [API] Add endpoint and handlers
+squash [API] Refactor replica repository interface
+squash [API] Add domain logic
+squash [API] Update in-mem repository implementation
+squash [API] Add sqlite repository support
+```
+
+### Larger projects
```

Comment:
[consider] Perhaps worth a comment on feature branches as a way to group multiple smaller PRs that individually leave the codebase in an undesirable (but still technically functional, based on the above description) state.

Reply:
Added https://github.com/TheFoundryVisionmongers/OpenAssetIO/issues/197 to add this soon.